lib/prompt.js

```diff
@@ -120,7 +120,18 @@ class Prompt extends Events {
       await this.options.onSubmit.call(this, this.name, this.value, this);
     }
 
-    let result = this.state.error || await this.validate(this.value, this.state);
+    let result;
+    try {
+      result =
+        this.state.error || (await this.validate(this.value, this.state));
+    } catch (e) {
+      this.state.cancelled = true;
+      await this.render();
+      await this.close();
+      this.emit("cancel", e);
+      return;
```

Comment:
I wanted to replace these lines with `this.cancel(e)` however due to the logic in `this.error(err)` the error is not propagated correctly.